include/pybind11/detail/common.h

```diff
@@ -386,7 +386,7 @@ PYBIND11_WARNING_DISABLE_CLANG("-Wgnu-zero-variadic-macro-arguments")
                 slots,                                                                            \
                 ##__VA_ARGS__);                                                                   \
         }();                                                                                      \
-        return result.ptr();                                                                      \
+        return PyModuleDef_Init(def);                                                             \
```

Comment:
For easy reference, this is the implementation of `PyModuleDef_Init()`:

https://github.com/python/cpython/blob/2fd09b011031f3c00c342b44e02e2817010e507c/Objects/moduleobject.c#L52-L62